.github/workflows/build.yml

```diff
@@ -1,19 +1,70 @@
-name: CI
+name: Build the book and deploy to github pages
 
 on:
-  pull_request:
   push:
+    branches: [ master, mdbook ]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Sets permissions of the GITHUB_TOKEN to allow deployment to GitHub Pages
+permissions:
+  contents: read
+  pages: write
+  id-token: write
+
+# Allow only one concurrent deployment, skipping runs queued between the run in-progress and latest queued.
+# However, do NOT cancel in-progress runs as we want to allow these production deployments to complete.
+concurrency:
+  group: "pages"
+  cancel-in-progress: false
+
 
 jobs:
   build:
     runs-on: ubuntu-latest
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v4
 
       - name: Install Nix
         uses: cachix/install-nix-action@v16
         with:
           nix_path: nixpkgs=channel:nixos-unstable
 
+      - name: Setup Pages
+        id: pages
+        uses: actions/configure-pages@v5
+
+      - name: Install mdbook
+        run: nix-env -iA mdbook mdbook-linkcheck -f '<nixpkgs>'
+
       - name: Build the book
-        run: nix-build release.nix
+        run: export PATH=~/.nix-profile/bin:$PATH && mdbook build
+      
+      - name: Install mdbook-epub
+        run: |
+          curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf -y | sh
+          rustup update
+          cargo install --version 0.4.37 mdbook-epub
+
+      - name: Generate epub
+        run: |
+          mdbook-epub --standalone true .
+          mv 'book/epub/Nix Pills.epub' book/html/nix-pills.epub
```

Comment:
Unfortunately this won't work. As mentioned before, we really need `nix-build release.nix -I nixpkgs=channel:nixos-unstable` to work, so that the homepage can include it in its Nix build pipeline. We cannot use impure installation methods there, so this needs to be turned into a Nix build.

Reply:
Sure! I wanted to have something that works, I agree it shouldn't be committed like this. I hope to learn enough Nix to make by myself `nix-build` work, but if not, I guess it wouldn't be hard for others to help me with this.

I added checkboxes in the PR description for remaining tasks.